### PR TITLE
Better test cases

### PR DESCRIPTION
**Pull request by damon-kwok**
https://github.com/ponylang/crypto/issues/31#issue-649875058
https://github.com/ponylang/crypto/issues/32#issue-649883999

This PR updates the test case: 
- Easier to expand
- Full coverage

```bash
1 test started, 0 complete: crypto/ConstantTimeCompare started
1 test started, 1 complete: crypto/ConstantTimeCompare complete
2 tests started, 1 complete: crypto/Hash/MD4 started
2 tests started, 2 complete: crypto/Hash/MD4 complete
3 tests started, 2 complete: crypto/Hash/MD5 started
3 tests started, 3 complete: crypto/Hash/MD5 complete
4 tests started, 3 complete: crypto/Hash/RIPEMD160 started
4 tests started, 4 complete: crypto/Hash/RIPEMD160 complete
5 tests started, 4 complete: crypto/Hash/SHA1 started
5 tests started, 5 complete: crypto/Hash/SHA1 complete
6 tests started, 5 complete: crypto/Hash/SHA224 started
6 tests started, 6 complete: crypto/Hash/SHA224 complete
7 tests started, 6 complete: crypto/Hash/SHA256 started
7 tests started, 7 complete: crypto/Hash/SHA256 complete
8 tests started, 7 complete: crypto/Hash/SHA384 started
8 tests started, 8 complete: crypto/Hash/SHA384 complete
9 tests started, 8 complete: crypto/Hash/SHA512 started
9 tests started, 9 complete: crypto/Hash/SHA512 complete
10 tests started, 9 complete: crypto/Digest/MD4 started
10 tests started, 10 complete: crypto/Digest/MD4 complete
11 tests started, 10 complete: crypto/Digest/MD5 started
11 tests started, 11 complete: crypto/Digest/MD5 complete
12 tests started, 11 complete: crypto/Digest/RIPEMD160 started
12 tests started, 12 complete: crypto/Digest/RIPEMD160 complete
13 tests started, 12 complete: crypto/Digest/SHA1 started
13 tests started, 13 complete: crypto/Digest/SHA1 complete
14 tests started, 13 complete: crypto/Digest/SHA224 started
14 tests started, 14 complete: crypto/Digest/SHA224 complete
15 tests started, 14 complete: crypto/Digest/SHA256 started
15 tests started, 15 complete: crypto/Digest/SHA256 complete
16 tests started, 15 complete: crypto/Digest/SHA384 started
16 tests started, 16 complete: crypto/Digest/SHA384 complete
17 tests started, 16 complete: crypto/Digest/SHA512 started
17 tests started, 17 complete: crypto/Digest/SHA512 complete
18 tests started, 17 complete: crypto/Digest/SHAKE128 started
18 tests started, 18 complete: crypto/Digest/SHAKE128 complete
19 tests started, 18 complete: crypto/Digest/SHAKE256 started
19 tests started, 19 complete: crypto/Digest/SHAKE256 complete
---- Passed: crypto/ConstantTimeCompare
---- Passed: crypto/Hash/MD4
---- Passed: crypto/Hash/MD5
---- Passed: crypto/Hash/RIPEMD160
---- Passed: crypto/Hash/SHA1
---- Passed: crypto/Hash/SHA224
---- Passed: crypto/Hash/SHA256
---- Passed: crypto/Hash/SHA384
---- Passed: crypto/Hash/SHA512
---- Passed: crypto/Digest/MD4
---- Passed: crypto/Digest/MD5
---- Passed: crypto/Digest/RIPEMD160
---- Passed: crypto/Digest/SHA1
---- Passed: crypto/Digest/SHA224
---- Passed: crypto/Digest/SHA256
---- Passed: crypto/Digest/SHA384
---- Passed: crypto/Digest/SHA512
---- Passed: crypto/Digest/SHAKE128
---- Passed: crypto/Digest/SHAKE256
----
---- 19 tests ran.
---- Passed: 19
```bash

### CODE REVIEW

**Comment by damon-kwok** [comment]
@SeanTAllen  mentioned:
> Either way, given that we have no tests that run in CI for this currently, any PR for this shouldn't be merged until #29 is done so that we can be sure that the tests are working and passing.

@SeanTAllen @mfelsche 
This PR solves #31 #32, it does not contradict #29, I think it can be previewed and merged.

File: crypto/_test.pony [hunk 1]
```diff
@@ -6,15 +6,50 @@ actor Main is TestList
...
-    test(_TestRIPEMD160)
-    test(_TestSHA1)
-    test(_TestSHA224)
-    test(_TestSHA256)
-    test(_TestSHA384)
-    test(_TestSHA512)
-    test(_TestDigest)
+
+    // test hash
+    test(_TestHash[MD4]("md4", "test", "db346d691d7acc4dc2625db19f9e3f52"))
```
Comment (SeanTAllen) [inline crypto/_test.pony:11]: I don't like this change. I think it makes it harder to see what failed, what is being run etc.

We can expand test coverage but I want it to follow the previous pattern.

File: crypto/_test.pony [hunk 2]
```diff
@@ -36,107 +71,62 @@ class iso _TestConstantTimeCompare is UnitTest
...
+  let _name: String
+  var d: (Digest | None) = None
+  let inputs: Array[String] val
+  let assert: String
+  
+  new iso create(name': String,
+    inputs': Array[String] val, assert': String)
+  =>
+    _name = "crypto/Digest/"+ name'.upper()
+    d =  match name'.lower()
```
Comment (SeanTAllen) [inline crypto/_test.pony:109]: this is an anti-pattern. this matching on a string name. We have a type system that can catch things at compile time. This moves errors to runtime. 

I would need a very very strong argument to ever move an error from being compile time to runtime.

**Comment by SeanTAllen** [comment]
Sorry, I'm not in favor of this PR.

**Pull request closed without merging**